addons/website_sale/tests/test_website_editor.py

```diff
@@ -0,0 +1,105 @@
+import odoo.tests
+from odoo.tests import HttpCase, new_test_user
+import odoo.exceptions
+
+GIF = b"R0lGODdhAQABAIAAAP///////ywAAAAAAQABAAACAkQBADs="  # Tiny blank base64 image data.
+ATTACHMENT_COUNT = 5
+
+@odoo.tests.tagged('post_install', '-at_install')
+class TestProductPictureController(HttpCase):
+    def setUp(self):
+        super().setUp()
+        self.user1 = new_test_user(self.env, "test_user_1", email="test_user_1@nowhere.com",
+                                   password="Password_test")
+        self.user2 = new_test_user(self.env, "test_user_2", email="test_user_2@nowhere.com",
+                                   password="Password_test",
+                                   groups='website.group_website_publisher,base.group_user,account.group_account_manager')
+        self.user3 = new_test_user(self.env, "test_user_3", email="test_user_3@nowhere.com",
+                                   password="Password_test",
+                                   groups='base.group_user,website.group_website_publisher')
+        self.product = self.env['product.product'].create({
+            'name': 'Storage Test Box',
+            'standard_price': 70.0,
+            'list_price': 79.0,
+            'website_published': True,
+        })
+
+        self.attachments = self.env['ir.attachment'].create([
+            {
+                'datas': GIF,
+                'name': f'image0{i}.gif',
+                'public': True
+            }
+            for i in range(ATTACHMENT_COUNT)])
```

Comment:
`Untested` : creating users is expensive, creating them in setUpClass will prevent every tests from creating the 4 users thus slowing down the total time to run your file.

Do note that there are legitimate cases where we want to do such things in `setUp` instead.

```suggestion
    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.user1 = new_test_user(cls.env, "test_user_1", email="test_user_1@nowhere.com",
                                   password="Password_test")
        cls.user2 = new_test_user(cls.env, "test_user_2", email="test_user_2@nowhere.com",
                                   password="Password_test",
                                   groups='website.group_website_publisher,base.group_user,account.group_account_manager')
        cls.user3 = new_test_user(cls.env, "test_user_3", email="test_user_3@nowhere.com",
                                   password="Password_test",
                                   groups='base.group_user,website.group_website_publisher')
        cls.product = cls.env['product.product'].create({
            'name': 'Storage Test Box',
            'standard_price': 70.0,
            'list_price': 79.0,
            'website_published': True,
        })

        cls.attachments = cls.env['ir.attachment'].create([
            {
                'datas': GIF,
                'name': f'image0{i}.gif',
                'public': True
            }
            for i in range(ATTACHMENT_COUNT)])
```